src/components/filters/checkbox.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC, ReactNode } from "react"
+
+import { formatNumber } from "../../lib/formattingHelper"
+
+import Checkbox from "../../components/checkbox"
+
+interface Item {
+  value: string | number
+  label: string
+  renderIcon?: () => ReactNode
+}
+
+interface Props {
+  name: string
+  options: Item[]
+  applyFilters: (filtersToApply) => void
+  selected?: (string | number)[]
+  facet?: { [key: string]: number }
+}
+
+const CheckboxFilter: FC<Props> = ({
+  name,
+  options,
+  applyFilters,
+  selected = [],
+  facet = {},
+  children,
+}) => {
+  const values = options.map(({ value }) => value)
+
+  const selectValue = (value) => () => {
+    const selectionCopy = [...selected]
+
+    if (selectionCopy.includes(value)) {
+      const i = selectionCopy.indexOf(value)
+      selectionCopy.splice(i, 1)
+    } else {
+      selectionCopy.push(value)
+    }
+
+    selectionCopy.sort((a, b) =>
```

Comment:
Why do they need to be sorted?

Reply:
Consider the following:

1. 
  - [ ] A
  - [x] B


2. 
  - [x] A
  - [x] B


3. 
  - [x] A 
  - [ ] B

Whan unsorted  going 1-2-3 would result in state changes `[]` -> `[B]` -> `[B, A]` -> `[A]`, while 3-2-1 would result in `[]` -> `[A]` -> `[A, B]` -> [B]` so the both filters applied would result in a different state. Given that this state can be used in various ways (e.g. to generate URL) I want to not care about the order the filter s are applied.

Reply:
I do get that the order of the selected array would vary based on the way the way the filters are applied I just wondered why it would matter for the application if we have something like `?filter=A,B` as a query wouldn't we get the same result if the query is `?filter=B,A`

Reply:
In terms of content - yes, in terms of things like browser caching - no since the URL would be different.